app/models/piece.rb

```diff
@@ -1,4 +1,34 @@
 class Piece < ApplicationRecord
   belongs_to :player
   belongs_to :game
+
+  def obstructed?(destination_x, destination_y)
+    x_difference = x_coordinate - destination_x
+    y_difference = y_coordinate - destination_y
+
+    @chess_in_between = []
+
+    # first, raise an argument error when self and sq have the same coordinates
+    if x_difference.zero? && y_difference.zero?
+      raise ArgumentError, 'destination has to have a different location'
+    elsif x_difference.zero?
+      # the two locations are on the same vertical line
+      @chess_in_between = Piece.where(x_coordinate: x_coordinate)
```

Comment:
`game.pieces.where`

Want to make sure we are only looking at pieces in the current game